[stable/nginx-ingress] add terminationGracePeriodSeconds option

henryoswald opened the pull request:
<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:
Adds terminationGracePeriodSeconds as a configurable option to nginx-ingress. Useful for allowing very long running requests to finish processing (websockets) before killing the pod. 


#### Which issue this PR fixes


#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

Commit:
[stable/nginx] add terminationGracePeriodSeconds option

Signed-off-by: Henry Oswald <henry.oswald@sharelatex.com>

ChiefAlexander commented on stable/nginx-ingress/Chart.yaml:
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: nginx-ingress
-version: 1.12.1
+version: 1.13.1
```
`1.13.0`

henryoswald replied:
Thanks, updated.

ChiefAlexander commented on stable/nginx-ingress/README.md:
```diff
@@ -79,6 +79,7 @@ Parameter | Description | Default
 `controller.daemonset.hostPorts.stats` | If `controller.daemonset.useHostPort` is `true` and this is non-empty, it sets the hostPort | `"18080"`
 `controller.tolerations` | node taints to tolerate (requires Kubernetes >=1.6) | `[]`
 `controller.affinity` | node/pod affinities (requires Kubernetes >=1.6) | `{}`
+`controller.terminationGracePeriodSeconds` | how many seconds to wait before terminating a pod | `{}`
```
default is 60 not `{}`

henryoswald replied:
Thanks, updated.

ChiefAlexander reviewed (changes requested):
Thanks for the PR! Just a few things to fixup

Commit:
[ci] set DaemonSet host ports to random ports

hopefully will help with ci timeouts
https://github.com/helm/charts/pull/16115#issuecomment-518814209

Signed-off-by: Henry Oswald <henry.oswald@sharelatex.com>
